### PR TITLE
fix(curriculum): allow any response status for error response

### LINKED ISSUE #37966: Validations http status code on APIs and Microservices Challenges
I've finished the timestamp microservice challenge, but I've found something that I think would be useful for students.

One of the user stories of the project says that:
> 5. the date string is invalid the api returns a JSON having the structure `{"error" : "Invalid Date" }`

I've submitted a solution returning the required JSON **and** also setting the 400 status code of the response and the judge rejected it, which I think (I have experience building APIs) its weird because you're telling the consumer in the JSON "hey, there is an error here" but also telling him on the http status "nothing bad, everything OK".

I think that in addition to returning the JSON in the aforementioned structure, the student should set the status code to 400 (bad request). That is to help students with the meaning of each http status code and to make them get used to use them; in real world APIs it's important to return the appropriate http status code if an error occurs.

If you think it's fine this feature, I can add a PR with this:
1. Update the user story to say that they should return the 400 status code and,
2. Update the judge to check http status codes.

If you think that it's not ok to update the user story and only require the JSON structure, I think it would be still useful to check status code on the judgement. I've spent 10 minutes figuring out what was the reason of the failure of my submission before I commented the 400 status code line in the code and getting and approval right after it.

My solution is on this repo: github.com/german1608/timestamp-microservice (without the `res.status(400)` line) and the glitch project is here https://oceanic-rifle.glitch.me

### PR DESCRIPTION
**Pull request by moT01**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #37966

Seemed reasonable to allow any response code when it's supposed to be an error.

### CODE REVIEW

**Review by ShaunSHamilton — COMMENTED** [review]
This makes sense, and I would be fine with this going as is, but I do find it useful when the errors are thrown to the console instead of silently asserted.

File: curriculum/challenges/english/05-apis-and-microservices/apis-and-microservices-projects/timestamp-microservice.md [hunk 1]
```diff
@@ -107,7 +107,7 @@ If the input date string is invalid, the api returns an object having the struct
       assert.equal(data.error.toLowerCase(), 'invalid date');
     },
     (xhr) => {
-      throw new Error(xhr.responseText);
+      assert(xhr.responseJSON.error.toLowerCase() === 'invalid date');
```
Comment (ojeytonwilliams) [inline curriculum/challenges/english/05-apis-and-microservices/apis-and-microservices-projects/timestamp-microservice.md:110]:
```suggestion
      assert(xhr.responseJSON.error.toLowerCase() === 'invalid date', 'received: ' + xhr.responseJSON.error);
```
How about this?  It should tell the user something if it's not the expected response.

Comment (moT01) [reply curriculum/challenges/english/05-apis-and-microservices/apis-and-microservices-projects/timestamp-microservice.md:110]: Where is this text showing up? I don't see it anywhere.

Comment (moT01) [reply curriculum/challenges/english/05-apis-and-microservices/apis-and-microservices-projects/timestamp-microservice.md]: Sure, lets just go with it

Comment (ojeytonwilliams) [reply curriculum/challenges/english/05-apis-and-microservices/apis-and-microservices-projects/timestamp-microservice.md]: Sorry, @moT01, I'm not sure why I thought this would work.  If we want to report to the user we'd need to do something like 

```
try {
  assert(xhr.responseJSON.error.toLowerCase() === 'invalid date');
} catch {
  console.error(xhr)
}
```
but I'm not sure that's necessary, given that they'll have error logs in the console when the status code is not 200.

TL;DR: let's revert my suggestion.

File: curriculum/challenges/english/05-apis-and-microservices/apis-and-microservices-projects/timestamp-microservice.md [hunk 2]
```diff
@@ -107,7 +107,7 @@ If the input date string is invalid, the api returns an object having the struct
       assert.equal(data.error.toLowerCase(), 'invalid date');
     },
     (xhr) => {
-      throw new Error(xhr.responseText);
+      assert(xhr.responseJSON.error.toLowerCase() === 'invalid date', 'received: ' + xhr.responseJSON.error);
```
Comment (ojeytonwilliams) [inline curriculum/challenges/english/05-apis-and-microservices/apis-and-microservices-projects/timestamp-microservice.md:110]:
```suggestion
      assert(xhr.responseJSON.error.toLowerCase() === 'invalid date');
```
Ooops.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1: sorry for the time wasting.

**Review by ShaunSHamilton — APPROVED** [review]
LGTM 👍
